namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ObserverCache.scala

```diff
@@ -55,13 +60,18 @@ class ObserverCache[K <: AnyRef, T](
   // ConcurrentHashMap is used to make reads lockless, but all updates are explicitly synchronized
   private[this] val activeCache = new ConcurrentHashMap[K, Observer[T]]
   private[this] val inactiveCache = CacheBuilder.newBuilder()
+    .ticker(new Ticker {
```

Comment:
This works but `Stopwatch.start()` results in some unnecessary allocations.  My suggestion of passing in an instance of Stopwatch as a parameter was so that by default we could use `Stopwatch.systemNanos` which is very performant and avoids unnecessary allocations but also allows us to pass in `Stopwatch.timeNanos` in the test which allows us to do time control.  These Stopwatch instances also return Long directly which allows us to avoid boxing and unboxing a Duration.

Reply:
@adleong, sorry for that. I didn't know this interface and I didn't understand you correctly. I updated my PR. Please check if everything is as you wanted.